File: api/notion.py
import json
import urllib.request
import urllib.error
import os

def handler(request):
    notion_key = os.environ.get("NOTION_API_KEY", "")
    headers = {
        "Authorization": f"Bearer {notion_key}",
        "Notion-Version": "2022-06-28",
        "Content-Type": "application/json"
    }

    def notion_req(method, endpoint, data=None):
        url = f"https://api.notion.com/v1{endpoint}"
        body = json.dumps(data).encode() if data else None
        req = urllib.request.Request(url, data=body, headers=headers, method=method)
        try:
            with urllib.request.urlopen(req) as r:
                return json.loads(r.read()), 200
        except urllib.error.HTTPError as e:
            return json.loads(e.read()), e.code

    cors = {
        "Access-Control-Allow-Origin": "*",
        "Access-Control-Allow-Methods": "GET, POST, OPTIONS",
        "Access-Control-Allow-Headers": "Content-Type",
        "Content-Type": "application/json"
    }

    if request.method == "OPTIONS":
        return Response("", 200, cors)

    path = request.path

    if request.method == "GET" and path == "/api/notion/databases":
        result, status = notion_req("POST", "/search", {
            "filter": {"value": "database", "property": "object"}
        })
        databases = []
        for item in result.get("results", []):
            t = item.get("title", [])
            title = t[0]["plain_text"] if t else "Sem título"
            databases.append({"id": item["id"], "title": title})
        return Response(json.dumps({"databases": databases}), status, cors)

    if request.method == "POST":
        body = json.loads(request.body or "{}")
        if path == "/api/notion/create-task":
            db_id = body.get("database_id")
            props = body.get("properties", {})
            result, status = notion_req("POST", "/pages", {
                "parent": {"database_id": db_id},
                "properties": props
            })
            return Response(json.dumps(result), status, cors)

        if path == "/api/notion/database-properties":
            db_id = body.get("database_id")
            result, status = notion_req("GET", f"/databases/{db_id}")
            props = result.get("properties", {}) if status == 200 else {}
            return Response(json.dumps({"properties": props}), status, cors)

    return Response(json.dumps({"error": "Not found"}), 404, cors)

File: api/notion.js
export default async function handler(req, res) {
  res.setHeader("Access-Control-Allow-Origin", "*");
  res.setHeader("Access-Control-Allow-Methods", "GET, POST, OPTIONS");
  res.setHeader("Access-Control-Allow-Headers", "Content-Type");

  if (req.method === "OPTIONS") return res.status(200).end();

  const NOTION_KEY = process.env.NOTION_API_KEY;
  const headers = {
    "Authorization": `Bearer ${NOTION_KEY}`,
    "Notion-Version": "2022-06-28",
    "Content-Type": "application/json"
  };

  const action = req.query.action;

  if (action === "databases") {
    const r = await fetch("https://api.notion.com/v1/search", {
      method: "POST",
      headers,
      body: JSON.stringify({ filter: { value: "database", property: "object" } })
    });
    const data = await r.json();
    const databases = (data.results || []).map(db => ({
      id: db.id,
      title: db.title?.[0]?.plain_text || "Sem título"
    }));
    return res.status(200).json({ databases });
  }

  if (action === "create-task") {
    const { database_id, properties } = req.body;
    const r = await fetch("https://api.notion.com/v1/pages", {
      method: "POST",
      headers,
      body: JSON.stringify({ parent: { database_id }, properties })
    });
    const data = await r.json();
    return res.status(r.status).json(data);
  }

  if (action === "db-properties") {
    const { database_id } = req.body;
    const r = await fetch(`https://api.notion.com/v1/databases/${database_id}`, { headers });
    const data = await r.json();
    return res.status(r.status).json({ properties: data.properties || {} });
  }

  return res.status(404).json({ error: "Not found" });
}
